tracing/src/main/java/com/palantir/tracing/Tracers.java

```diff
@@ -369,6 +369,30 @@ public static Runnable wrapWithNewTrace(String operation, Observability observab
         };
     }
 
+    /**
+     * Wraps the given {@link Callable} such that it creates a fresh {@link Trace tracing state with the given traceId}
+     * for its execution. That is, the trace during its {@link Callable#call() execution} will use the traceId provided
+     * instead of any trace already set on the thread used to execute the callable. Each execution of the callable
+     * will use a new {@link Trace tracing state} with the same given traceId. The given {@link String operation} is
+     * used to create the initial span.
+     */
+    public static <V> Callable<V> wrapWithAlternateTraceId(String traceId, String operation,
+            Observability observability, Callable<V> delegate) {
```

Comment:
This could introduce an abiguity at the compilation level between this and the override which takes a Runnable, but I've verified that there are only 3 invocations internally and so we should be fine.